tests/worker/test_reworker.py

```diff
@@ -162,7 +162,8 @@ def test_clear_task(worker: Worker) -> None:
 
 
 def test_clear_nonexistant_task(worker: Worker) -> None:
-    assert not worker.clear_task("foo")
+    with pytest.raises(KeyError):
+        worker.clear_task("foo")
```

Comment:
@tpoliaw may have views on this since I had this way originally and he preferred to return a bool

Reply:
This way is slightly neater to make use of the exceptionHandler was my rationale

Reply:
My main reservation about it was when there were discussions of pending tasks could expire and be automatically removed. It didn't make sense for clearing a task that had already been removed to be an error. I'm not strongly against either approach although would err on the side of calling clear multiple times being a no-op instead of an error.

Reply:
Per https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time I'm recommending sticking with a 404 when deleting a resource that doesn't exist.